### PR TITLE
docs: using typedoc to generate api documentation

### LINKED ISSUE #5780: publish api documentation to the website
### Is your feature request related to a problem? Please describe

it should be the first step, publishing API doc at a location on podman-desktop.io website

### Describe the solution you'd like

publish to like `/api/reference` the content of the typedoc content

### Describe alternatives you've considered

it will be a step 0, will need to have a curated page after that.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

Adding an **hidden** `/api` documentation for the generated documentation.

### Screenshot / video of UI

image

### What issues does this PR fix or reference?

fixes https://github.com/containers/podman-desktop/issues/5780

### How to test this PR?

- Checkout the branch
- CD inside the `website` folder and run `yarn run start`
- Visit `localhost:3000/api`

### CODE REVIEW

File: website/docusaurus.config.js [hunk 1]
```diff
@@ -309,6 +309,27 @@ const config = {
...
+        path: 'api',
+        routeBasePath: 'api',
+      },
+    ],
+    [
+      'docusaurus-plugin-typedoc',
+      {
+        id: 'api',
+        plugin: ['typedoc-plugin-markdown'],
+        entryPoints: ['..\\packages\\extension-api\\src\\extension-api.d.ts'],
```
Comment (benoitf) [inline website/docusaurus.config.js]: FYI the path is not working on macOS (it says path not found)

if slashes are not working on windows, probably uses javascript with ${path.sep}

File: packages/extension-api/src/extension-api.d.ts [hunk 2]
```diff
@@ -2920,8 +2920,7 @@ declare module '@podman-desktop/api' {
     /**
      * Store a new value for key in the context.
      * This can be used in enablement of command or with the when property.
-     * The key should consists of '<extension-
-     * id>.<actual-key>'.
+     * The key should consists of '"extension-id"."actual-key"'.
```
Comment (benoitf) [inline packages/extension-api/src/extension-api.d.ts:2982]: hint: could have been a separated PR saying that it'll cause troubles if exporting to html

File: website/docusaurus.config.js (hunk 1 above)
Comment (axel7083) [reply website/docusaurus.config.js]: I replaced with `require.resolve`

Comment (benoitf) [reply website/docusaurus.config.js]: require is kind of forbidden these days

could you add

import {resolve} from 'node:path';

and then use resolve('') (without require prefix)

**Comment by benoitf** [comment]
please use `node:path` import

also it seems it's raising an error when building the project (not development mode)

**Comment by axel7083** [comment]
> please use `node:path` import
> 
> also it seems it's raising an error when building the project (not development mode)

Yeah, I'm investigating, I was able to fix it when using the next version of `typedoc-plugin-markdown`, seems to be an issue on their side

**Comment by axel7083** [comment]
Okey, after an investigation the plugin is under large re-work, I asked the author for a release date (https://github.com/tgreyuk/typedoc-plugin-markdown/issues/552) he stated that the current `1.0.0-next.25` is `pretty much feature complete`.

**Comment by benoitf** [comment]
almost done 🎉 

we need to filter out screenshots for the API (it's adding 265 screenshots (too much))
and only select a subset of the pages (as it's generated) like one namespace, one class, etc.

we're really close

for those wanted to browse the results:
https://65b7e37ed710302906fe2b64--podman-desktop-pr.netlify.app/api

**Comment by deboer-tim** [comment]
That is awesome, great work!  🎉

**Comment by axel7083** [comment]
I've open an issue on the `docusaurus-plugin-typedoc` repository about the sidebar, and he fixed it in a new release. I bumped the version !

**Comment by axel7083** [comment]
See current version of the generated documentation here https://65ba53ed088606dcf682ee61--podman-desktop-pr.netlify.app/

**Review by benoitf — APPROVED** [review]
👍

**Review by deboer-tim — APPROVED** [review]

**Pull request merged**